tests/staggered_dslash_test_utils.h

```diff
@@ -0,0 +1,379 @@
+#pragma once
+
+using namespace quda;
+
+#define staggeredSpinorSiteSize 6
+
+dslash_test_type dtest_type = dslash_test_type::Dslash;
+CLI::TransformPairs<dslash_test_type> dtest_type_map {{"Dslash", dslash_test_type::Dslash},
+                                                      {"MatPC", dslash_test_type::MatPC},
+                                                      {"Mat", dslash_test_type::Mat}
+                                                      // left here for completeness but not support in staggered dslash test
+                                                      // {"MatPCDagMatPC", dslash_test_type::MatPCDagMatPC},
+                                                      // {"MatDagMat", dslash_test_type::MatDagMat},
+                                                      // {"M5", dslash_test_type::M5},
+                                                      // {"M5inv", dslash_test_type::M5inv},
+                                                      // {"Dslash4pre", dslash_test_type::Dslash4pre}
+                                                    };
+
+struct DslashTime {
+  double event_time;
+  double cpu_time;
+  double cpu_min;
+  double cpu_max;
+
+  DslashTime() : event_time(0.0), cpu_time(0.0), cpu_min(DBL_MAX), cpu_max(0.0) {}
+};
+
+struct StaggeredDslashTestWrapper {
```

Comment:
Can you add a boolean/`QudaBoolean` field to this that specifies if it's being called from `staggered_dslash_ctest` or `staggered_dslash_test`? I'll use this to re-introduce caching the gauge fields later.